### ISSUE TITLE
New feature multipart - how to use?

### ISSUE THREAD

**Issue by WSorban** [silkimen/cordova-plugin-advanced-http#273]
I was using Angular's HttpClient to post some multipart data, as follows:

```
        const headers = {
            headers: new HttpHeaders({
                enctype: 'multipart/form-data'
            })
        };

        this.httpClient.post(`${routes.flashAirPost()}`, formData, headers).toPromise()
```

The problem is that with this, I encounter CORS issues. In the browser, things are so far so good (either using a CORS proxy, or a browser plugin). I have no means of modifying the server.

The app I'm making is supposed to be used from my Android device.

Therefore, I stumbled upon your Cordova Plugin, and to my surprise, `multipart` has been developed recently.

Despite not yet being released to NPM repository, I have started using it.

`http.setDataSerializer('multipart');`
`http.post(`${routes.flashAirPost()}`, formData, {enctype: 'multipart/form-data'}, fnSuccess, fnError);`

For some reason, the server is refusing this request.

Would you be able to confirm that what I did is equivalent of the HttpClient request code?
HttpClient requests can be visualized from the browser's Network tool.

How could I debug Advanced Http's requests?

Edit:
Is there something else that needs to be done?

> ⚠️ multipart depends on several Web API standards which need to be supported in your web view. Check out https://github.com/silkimen/cordova-plugin-advanced-http/wiki/Web-APIs-required-for-Multipart-requests for more info.

(The page contains nothing at the moment)

**Comment by silkimen** [comment]
Hi Wihelm, this feature is not released yet, because it lacks some documentation (you've realized that already) and it needs to be tested on some real devices (was tested on emulators and one real device only). Do you see some warnings or error messages in your developer console when running your code on the device?
Your Webview needs to implement `FormData.prototype.entries` and it needs to support `FileReader` API and `TextEncoder` API.
If `FormData.prototype.entries` is not implemented, you can also use a ponyfill which is shipped with this plugin e.g. `new cordova.plugin.http.ponyfill.FormData()`. But be aware that these FormData instances are meant to be used for this plugin only. If you try to submit it via XMLHttpRequest or fetch API, it won't work.

**Comment by WSorban** [comment]
Thanks!
Perhaps this is the infor I was missing!
Unfortunately, tomorrow I will need to travel for a week, therefore I will not be able to do further tests.

For now, there are no warnings or error messages in the console, regarding the Webview.

Regarding the server, I am getting a response saying that the request could not be processed, meaning that most likely some headers, or some other stuff are wrong.

I posted the first case for reference, which is accepted by the server, just fine, if running from a browser, with a CORS proxy.

So basically, there isn't or it shouldn't be anything special.

**Comment by silkimen** [comment]
Closing this, feel free to open a new issue, if this won't solve your problem! 👍
